lib/data/sender.js

```diff
@@ -46,7 +46,9 @@ DataTrackSender.prototype.addDataChannel = function addDataChannel(dataChannel)
  * @returns {this}
  */
 DataTrackSender.prototype.removeDataChannel = function removeDataChannel(dataChannel) {
-  this._dataChannels.delete(dataChannel);
+  if (this._dataChannels.delete(dataChannel)) {
+    dataChannel.close();
+  }
```

Comment:
I was thinking about this some more—can we move this responsibility to PeerConnectionV2's `removeDataTrackSender`? e.g.,

```
PeerConnectionV2.prototype.removeDataTrackSender = function removeDataTrackSender(dataTrackSender) {
  var dataChannel = this._dataTrackSenders.get(dataTrackSender);
  if (dataChannel) {
    dataTrackSender.removeDataChannel(dataChannel);
    this._dataTrackSenders.delete(dataTrackSender);
  }
};
```

Reply:
Just to elaborate on my thinking here: PeerConnectionV2 currently creates the RTCDataChannel, but if we let DataTrackSender `close` the RTCDataChannel, now we have the responsibility for its lifecycle split over two classes. Would rather let PeerConnectionV2 manage its lifecycle and DataTrackSender just call `send`.

Reply:
@markandrus I was thinking that once PeerConnectionV2 creates the RTCDataChannel, it transfers the ownership to DataTrackSender. This design pattern is pretty common in C++ for ex: `scoped_ptr`, which is assigned some memory, but when execution goes out of the scope in which it was created, the `scoped_ptr` now automatically de-allocates the memory associated with it.